cpp/ql/test/query-tests/Security/CWE/CWE-089/SqlTainted/test.cpp

```diff
@@ -0,0 +1,48 @@
+int sprintf(char* str, const char* format, ...);
+
+namespace std
+{
+	template<class charT> struct char_traits;
+
+	template <class T> class allocator {
+	public:
+		allocator() throw();
+	};
+	
+	template<class charT, class traits = char_traits<charT>, class Allocator = allocator<charT> >
+	class basic_string {
+	public:
+		explicit basic_string(const Allocator& a = Allocator());
+		basic_string(const charT* s, const Allocator& a = Allocator());
+
+		const charT* c_str() const;
+	};
+
+	typedef basic_string<char> string;
+}
+
+namespace pqxx {
+    struct connection {};
+
+    struct row {};
+    struct result {};
+
+    struct work {
+      work(connection&);
+
+      row exec1(const char*);
+      result exec(const std::string&);
+      std::string quote(const char*);
+    };
+}
+
+int main(int argc, char** argv) {
+    pqxx::connection c;
+    pqxx::work w(c);
+    
+    pqxx::row r = w.exec1(argv[1]); // BAD
+    
+    pqxx::result r2 = w.exec(w.quote(argv[1])); // GOOD
```

Comment:
The commit comment where this was introduced says "Move the example from the experimental CWE-089 query into a test".  Where was this exactly?  Or is it essentially a new test?

Reply:
Oh, good point. It's a "new" test that I've created by simplifying (and stubbing) the example in https://github.com/github/codeql/pull/6133/commits/2e2673aff649efbf04145899e035c08b6946e745.

Reply:
Great.